pandas/_testing.py

```diff
@@ -1456,7 +1456,19 @@ def assert_series_equal(
             check_dtype=check_dtype,
             index_values=np.asarray(left.index),
         )
-    elif needs_i8_conversion(left.dtype) or needs_i8_conversion(right.dtype):
+    elif (
+        is_extension_array_dtype(left.dtype) and needs_i8_conversion(right.dtype)
+    ) or (is_extension_array_dtype(right.dtype) and needs_i8_conversion(left.dtype)):
+        # If we have an extension array and Datetimearray / TimedeltaArray, we still
+        # want to assert_extension_array_equal even though Datetimearray and
```

Comment:
this is not correct, the *dtypes* of a DTA or TDA are not EA dtypes (well except a tz-aware DTA).

```
    >>> is_extension_type(pd.DatetimeIndex([1, 2, 3]))
    False
    >>> is_extension_type(pd.DatetimeIndex([1, 2, 3], tz="US/Eastern"))
    True
```

the check is ok, its the text that needs updating.

Reply:
Hey @jreback ! 
Thanks for the quick response. Are you saying that just the comment needs to be changed? 
Maybe to: 

> If we have an extension array dtype and Datetime dtype / Timedelta dtype..

I'm a bit confused otherwise because I say later in the comment that 

> Datetimearray and TimedeltaArray dtypes are _not_ an instance of ExtensionArraydtype subclass

Reply:
see my comment again. its not about DTA or TDA themselve, rather their *dtypes*

Reply:
Would this suffice?
> If we have an array of dtype EA and a DTA/TDA (whose dtype is not of EA), we still perform assert_extension_array_equal